R/pkg/R/functions.R

```diff
@@ -1332,6 +1332,48 @@ setMethod("quarter",
             column(jc)
           })
 
+#' @details
+#' \code{percentile_approx} Returns the approximate percentile value of
+#' numeric column at the given percentage.
+#'
+#' @param percentage numeric percentage at which percentile should be computed
+#'                   All values should be between 0 and 1.
+#'                   If length equals to 1 resulting column is of type double,
+#'                   otherwise array<double>/
+#' @param accuracy parameter (default: 10000) is a positive numeric literal which
+#' controls approximation accuracy at the cost of memory. Higher value of accuracy
+#' yields better accuracy, 1.0/accuracy is the relative error of the approximation.
+#'
+#' @rdname column_aggregate_functions
+#' @aliases percentile_approx percentile_approx,Column-method
+#' @note percentile_approx since 3.0.0
+setMethod("percentile_approx",
+          signature(x = "characterOrColumn", percentage = "numeric", accuracy = "numeric"),
+          function(x, percentage, accuracy) {
```

Comment:
I think you can remove `accuracy` from the signature, and assign the default value `accuracy=10000L`.

Reply:
This way is more consistent and you can only define one method.

Reply:
I think this will requires switching to ellipsis in generic:

```r
setGeneric("percentile_approx",  function(x, percentage, ...)  { standardGeneric("percentile_approx") })

setMethod("percentile_approx",
          signature(x = "characterOrColumn", percentage = "numeric"),
          function(x, percentage, accuracy = 10000) {
             ...
```

or `ANY` in a `signature`

```r
setGeneric("percentile_approx",
           function(x, percentage, accuracy = 10000) { standardGeneric("percentile_approx") })

setMethod("percentile_approx",
          signature(x = "characterOrColumn", percentage = "numeric", accuracy = "ANY"),
          function(x, percentage, accuracy = 10000) {
             ...
```

and none seemed very precise, but I don't have strong opinion about it. Maybe the latter.